tools/tesseratos/src/tesseratos/metrics_panel/plugin.cpp

```diff
@@ -0,0 +1,38 @@
+#include <imgui.h>
+
+#include <cubos/engine/imgui/plugin.hpp>
+#include <cubos/engine/imgui/serialization.hpp>
+
+#include <tesseratos/metrics_panel/plugin.hpp>
+
+using cubos::core::ecs::Read;
+using cubos::engine::Cubos;
+using cubos::engine::DeltaTime;
+
+using namespace tesseratos;
+
+struct Metrics
+{
+    double time_elapsed = 0.0;
+    double fps = 0.0;
+};
+
+Metrics metrics;
+
+static void panel(Read<DeltaTime> deltatime)
+{
+    ImGui::Begin("Metrics Panel");
+    metrics.time_elapsed += deltatime->value;
+    metrics.fps = 1 / deltatime->value;
+    ImGui::Text("Time: %.2f", metrics.time_elapsed);
+    ImGui::Text("FPS: %.2f", metrics.fps);
+
+    ImGui::End();
+}
+
+void tesseratos::metricsPanelPlugin(Cubos& cubos)
+{
+    cubos.addPlugin(cubos::engine::imguiPlugin);
+
+    cubos.system(panel).tagged("cubos.imgui");
```

Comment:
Dont declare global variables. The correct way of using `Metrics` is to add it to `cubos` object and then accessing it in your system via `Read<Metrics>` (if you want read permissions) or `Write<Metrics>` for read/write permissions.

Reply:
(You can check how the others plugins do it aswell)